src/routes/docs/products/ai/integrations/fal-ai/+page.markdoc

```diff
@@ -0,0 +1,216 @@
+    <div class="top-cover u-padding-block-end-56">
+      <div class="container">
+        <div class="u-flex u-gap-16 u-flex-justify-center u-margin-block-start-16">
+          <h1 class="heading-level-1">fal.ai Demo</h1>
+          <code class="u-un-break-text"></code>
+        </div>
+        <p class="body-text-1 u-normal u-margin-block-start-8" style="max-width: 50rem">
+          Use this page to test your implementation with FAL. Enter
+          text and receive the model output as a response.
+        </p>
+      </div>
+    </div>
+    <div class="container u-margin-block-start-negative-56"
+      x-data="{ prompt: '', result: { src: '' }, loading: false }">
+      <div class="card u-flex u-gap-24 u-flex-vertical">
+        <div class="u-flex u-cross-center u-gap-8">
+          <div class="input-text-wrapper is-with-end-button u-width-full-line">
+            <input x-model="prompt" type="search" placeholder="Prompt" />
+            <div class="icon-search" aria-hidden="true"></div>
+          </div>
+          <button class="button" x-bind:disabled="loading"
+            x-on:click="async () => { loading = true; result = { src: '' }; try { result = await onSubmit(prompt) } catch(err) { console.error(err); } finally { loading = false; } }">
+            <span class="text">Generate</span>
+          </button>
+        </div>
+        <template x-if="answer.type">
+          <div class="u-flex u-flex-vertical u-gap-12">
+            <div class="u-flex u-flex-vertical u-gap-12 card">
+              <div class="u-flex u-gap-12">
+                <h5 class="eyebrow-heading-2">Result:</h5>
+              </div>
+              <img class="u-max-width-400" x-bind:src="result.src" alt="FAL output" />
+            </div>
+          </div>
+        </template>
+      </div>
+    </div>
+  </main>
+</body>
+```
+
+All of this together will render a form that will submit your question to the Appwrite Function through a POST request which we'll create next. The Appwrite Function will call fal.ai's API and return the response, which will be displayed on your page using different conditional statements depending on the output media type.
+{% /section %}
+
+{% section #step-6 step=6 title="Handle POST Request" %}
+Now that we're serving a basic HTML page, we can add methods necessary to integrate with fal.ai's API.
+
+Import the fal.ai SDK at the top of our `main.js` file:
+
+```js
+import * as fal from '@fal-ai/serverless-client';
+```
+
+After handling the `GET` requests, initialize the fal.ai SDK:
+
+```js
+fal.config({ credentials: process.env.FAL_API_KEY });
+```
+
+Next, make the request to generate an image using the SDXL model:
+```js
+try {
+  const result = await fal.subscribe('fal-ai/fast-sdxl', {
+    input: {
+      prompt: req.body.prompt,
+    },
+  });
+  return res.json({ ok: true, src: result.images[0].url });
+} catch (e) {
+  error(e);
+  return res.json({ ok: false, error: 'Failed to generate image' }, 500);
+}
+```
```

Comment:
So this snippet is fine, but it doesn't talk about where it needs to be added within our main export function we already defined. Would be better to show where this code needs to be added in context of the GET request we already created above

Reply:
Simplified this section